.github/workflows/deploy.yml

```diff
@@ -0,0 +1,42 @@
+name: Deploy
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sendgrid-csharp
+        uses: actions/checkout@v2
+
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1.8.2
+        with:
+          dotnet-version: '3.1.x'
+
+      - name: Publish package to NuGet
+        run: |
+          make release
+          dotnet nuget push **/*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json --skip-duplicate
+
+  notify-on-failure:
+    name: Slack notify on failure
+    if: ${{ failure() }}
+    needs: [ deploy ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_COLOR: 'danger'
+          SLACK_ICON_EMOJI: ':github:'
+          SLACK_MESSAGE: ${{ format('Failed to deploy {1}{3} {0}/{1}/actions/runs/{2}', github.server_url, github.repository, github.run_id, ':') }}
+          SLACK_TITLE: Deployment Failure
+          SLACK_USERNAME: GitHub Actions
+          SLACK_MSG_AUTHOR: twilio-dx
+          SLACK_FOOTER: Posted automatically using GitHub Actions
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
```

Comment:
Is there a particular reason that we are sending deployment failures slack messages to #alerts-dev-interfaces instead of #deploys-dev-interfaces?

Reply:
#deploys- was intended to be created as part of the librarian migration for notifications around Admiral deployments. Calling these files "deploy" is a smidge misleading, as they more refer to asset publishing aspect of the release process. Nothing is really getting deployed, per se

Reply:
I named the very first workflow file in the GitHub Actions migration "deploy.yml" because it was a conversion of the "deploy:" field in the .travis.yml configuration. I think all these `deploy.yml` files should really be named `release.yml` or something similar, but hindsight's always like that...

Reply:
Since we have the opportunity I think we should go ahead and rename the files for all the Sendgrid repos to `release.yml` since this is the first Sendgrid Repo we are migrating from travis to GH. It's not a big enough deal to change in the Twilio repos... Maybe we can just add the renaming of the deploy.yml to the tickets where we are adding in sonar cloud analysis.

Reply:
I think it'll be moot when we implement the outcome from the no waiting spike, since that intends to combine the test and release workflows into a single one

Reply:
but yeah I can rename these to release